### PR TITLE
Revise generic wallet tests

### LINKED ISSUE #224: Restructure and extend generic wallet tests
### Location

wallet/test, backend/.../wallet

### Problem

The wallet tests appear a bit outdated and unstructured. To prepare additional backends, we should consider restructuring and extending the generic wallet tests.

### Proposal

We try to apply the following guidelines:
- Try to have one test file for each interface type in the package.
- Include the name of the interface that is being tested in the test function name.

### PR DESCRIPTION
**Pull request by matthiasgeihs**
Closes #224

### CODE REVIEW

**Comment by matthiasgeihs** [comment]
After some discussion, we decided that revising the generic tests needs more work. We decided to resume work on it at a later point in time.

**Comment by matthiasgeihs** [comment]
Update: Some changes from this PR (in particular the introduction of ZeroAddress on wallet test Setup) are needed for the upcoming backend implementations.

File: wallet/test/wallet.go [hunk 1]
```diff
@@ -35,6 +35,7 @@ type Setup struct {
 	Backend         wallet.Backend // backend implementation
 	Wallet          wallet.Wallet  // the wallet instance used for testing
 	AddressInWallet wallet.Address // an address of an account in the test wallet
+	ZeroAddress     wallet.Address // an address that is less or equal to any other address
```
Comment (ggwpez) [inline wallet/test/wallet.go:38]: Why less **or** equal? An address should never be equal to anything but itself.

File: wallet/test/wallet.go [hunk 2]
```diff
@@ -34,39 +32,17 @@ type UnlockedAccount func() (wallet.Account, error)
...
-	// Signature tests
-	DataToSign []byte
+	Backend        wallet.Backend // backend implementation
+	Wallet         wallet.Wallet  // the wallet instance used for testing
+	Address        wallet.Address // an address of an account in the test wallet
+	DataToSign     []byte         // some data to sign
+	AddressEncoded []byte         // a valid nonzero address not in the wallet
 }
 
-type NewSetupFunc func(rng *rand.Rand) *Setup
```
Comment (ggwpez) [inline wallet/test/wallet.go]: If i see it correctly, you added this function in cc85cc9fd333cc6645aa5f8340320fb6475748ec.  
So maybe you can fix it up.

**Review by ggwpez — COMMENTED** [review]
LGTM

File: wallet/test/wallet.go (hunk 1 above)
Comment (matthiasgeihs) [reply wallet/test/wallet.go:38]: A zero address can be equal to another zero address.

Comment (ggwpez) [reply wallet/test/wallet.go:38]: Ok, so there can be multiple zero addresses.

**Comment by matthiasgeihs** [comment]
Github Actions seems to be not working correctly. The CI was not triggered after [force push](https://github.com/hyperledger-labs/go-perun/compare/0dbb06af450953146eadd50761f049c8642362ba..767a1e32bf203f13d0ac17b474c0e82cb8181ad6). Even a second force push (without changes) did not help.

**Comment by matthiasgeihs** [comment]
Close and reopen also did not trigger the CI.

**Comment by matthiasgeihs** [comment]
The GitHub Actions service is currently degraded: https://www.githubstatus.com

I'll wait for the service to be up and running and will check back later.

**Review by ggwpez — APPROVED** [review]
CI finally worked :laughing:

**Pull request merged**